iron-menu-behavior.html

```diff
@@ -148,7 +148,12 @@
         var item = this.items[(curFocusIndex - i + length) % length];
         if (!item.hasAttribute('disabled')) {
           this._setFocusedItem(item);
-          return;
+
+          // Focus might not have worked, if the element was hidden or not
+          // focusable. In that case, try again.
+          if (document.activeElement == item) {
```

Comment:
I tested this it works fine with both shadow/shady dom. I think what @danbeam pointed out would be the case if element that gets focus in the element's shadow tree.